upf/upf_proxy.c

```diff
@@ -364,6 +364,12 @@ proxy_start_connect_fn (const u32 * session_index)
 
   /* TODO: RACE: should not access the flow on the main thread */
   flow = pool_elt_at_index (fm->flows, ps->flow_index);
+  if (pool_is_free_index (gtm->sessions, flow->session_index))
```

Comment:
This does not look exactly like a fix, isn't that more of a workaround? The idea which was mentioned in the comment above is that with a separate _single_ worker thread, we should only access the flowable from that worker thread. With multiple workers, things become harder as we have concurrent access to the flowtable from multiple workers. So perhaps we should add a comment that we still have a race here, and create a follow-up issue for proper multithreaded flowtable handling?